server/channels/api4/report.go

```diff
@@ -85,3 +88,46 @@ func getUsersForReporting(c *Context, w http.ResponseWriter, r *http.Request) {
 		c.Logger.Warn("Error writing response", mlog.Err(jsonErr))
 	}
 }
+
+func startUsersBatchExport(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !(c.IsSystemAdmin() && c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadUserManagementUsers)) {
+		c.SetPermissionError(model.PermissionSysconsoleReadUserManagementUsers)
+		return
+	}
+
+	if err := c.App.StartUsersBatchExport(c.AppContext, r.URL.Query().Get("date_range")); err != nil {
+		c.Err = err
+		return
+	}
+
+	ReturnStatusOK(w)
+}
+
+func retrieveBatchReportFile(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Why do we have a discrete API to fetch this file and not just include it as an attachment to direct message we send the user? Would that also absolve us of the need to maintain a custom post type?

Reply:
Could we use attachment permalinks here to share with other admins instead? 
![CleanShot 2024-01-11 at 13 18 48@2x](https://github.com/mattermost/mattermost/assets/1023171/6aedf3d1-8f21-4326-94ff-0734edef5d2f)

Reply:
Does that not make it public then? That's something we want to avoid I assume right now.

Reply:
Yeah, fair enough. I'd be curious if we can reuse the same file info in multiple channels.

Reply:
I think the easiest way is to just have admins share the link for now, but as suggested above by Harshil we may want to have some system to distribute to multiple users, but that's a future concern for now.